[sdk 6.0.100-preview.1.21107.2-win-x64] dotnet tool fail due to argument passed to the tool.

The cli tries to parse the '-v' argument that is passed to the tool, previously the cli ignored this argument.
```
25>System.InvalidOperationException: Cannot parse argument 'Release' as expected type Microsoft.DotNet.Cli.VerbosityOptions.
25>   at System.CommandLine.Binding.ArgumentConverter.GetValueOrDefault[T](ArgumentConversionResult result)
25>   at System.CommandLine.Parsing.OptionResultExtensions.GetValueOrDefault[T](OptionResult optionResult)
25>   at System.CommandLine.Parsing.ParseResult.ValueForOption[T](String alias)
25>   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.LogVerbosityForAllTopLevelCommand(ICollection`1 result, ParseResult parseResult, String topLevelCommandName)
25>   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.Filter(Object objectToFilter)
25>   at Microsoft.DotNet.Cli.Utils.TelemetryEventEntry.SendFiltered(Object o)
25>   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
25>   at Microsoft.DotNet.Cli.Program.Main(String[] args)
25>C:\dev\Opal\CKSetup\CKSetup.csproj(12,5): error MSB3073: The command "dotnet cksetup run CKSetup.xml -v Release --logFile LastRun.log" exited with code 1.
```
I believe this is a regression from .NET 5.

> C:\dev\Opal\CKSetup\CKSetup.csproj(12,5): error MSB3073: The command "dotnet cksetup run CKSetup.xml -v Release --logFile LastRun.log" exited with code 1.

@Kuinox I believe this is a problem with the way you're attempting to run the tool, if the tool has been installed globally and does not have the `dotnet-` prefix in the name you don't have to specify `dotnet` before your command. "cksetup run CKSetup.xml -v Release --logFile LastRun.log" seems to work fine for me.

--- a/src/Cli/dotnet/CommandLineValidation/CommandLineValidationMessages.cs
+++ b/src/Cli/dotnet/CommandLineValidation/CommandLineValidationMessages.cs
@@ -72,6 +72,7 @@ namespace Microsoft.DotNet.Cli
                 CommandResult commandResult => commandResult.Token,
                 OptionResult optionResult => optionResult.Token ??
                                              new Token($"--{optionResult.Option.Name}", TokenType.Option),
+                ArgumentResult argResult => new Token(argResult.GetValueOrDefault<string>(), TokenType.Argument),
                 _ => null
             };
         }

--- a/src/Cli/dotnet/ParseResultExtensions.cs
+++ b/src/Cli/dotnet/ParseResultExtensions.cs
@@ -19,9 +19,15 @@ namespace Microsoft.DotNet.Cli
         {
             if (parseResult.Errors.Any())
             {
-                throw new CommandParsingException(
-                    message: string.Join(Environment.NewLine,
-                                         parseResult.Errors.Select(e => e.Message)));
+                var unrecognizedTokenErrors = parseResult.Errors.Where(error => 
+                    error.Message.Contains(Parser.Instance.Configuration.ValidationMessages.UnrecognizedCommandOrArgument(string.Empty).Replace("'", string.Empty)));
+                if (parseResult.CommandResult.Command.TreatUnmatchedTokensAsErrors ||
+                    parseResult.Errors.Except(unrecognizedTokenErrors).Any())
+                {
+                    throw new CommandParsingException(
+                        message: string.Join(Environment.NewLine,
+                                             parseResult.Errors.Select(e => e.Message)));
+                }
             }
             else if (parseResult.HasOption("--help"))
             {

--- a/src/Cli/dotnet/ParseResultExtensions.cs
+++ b/src/Cli/dotnet/ParseResultExtensions.cs
@@ -63,6 +63,10 @@ namespace Microsoft.DotNet.Cli
             {
                 return symbolResult.Symbol.Name;
             }
+            else if (symbolResult.Token().Type.Equals(TokenType.Argument))
+            {
+                return symbolResult.Token().Value;
+            }
             else
             {
                 return string.Empty;

--- a/src/Cli/dotnet/Telemetry/TelemetryFilter.cs
+++ b/src/Cli/dotnet/Telemetry/TelemetryFilter.cs
@@ -117,7 +117,7 @@ namespace Microsoft.DotNet.Cli.Telemetry
             ParseResult parseResult,
             string topLevelCommandName)
         {
-            if (parseResult.HasOption("--verbosity"))
+            if (parseResult.IsDotnetBuiltInCommand() && parseResult.HasOption("--verbosity"))
             {
                 result.Add(new ApplicationInsightsEntryFormat(
                     "sublevelparser/command",

--- a/src/Tests/dotnet.Tests/ParserTests/ToolRunParserTests.cs
+++ b/src/Tests/dotnet.Tests/ParserTests/ToolRunParserTests.cs
@@ -55,5 +55,12 @@ namespace Microsoft.DotNet.Tests.ParserTests
             result.UnparsedTokens.Should().Contain("hi2");
             result.ShowHelpOrErrorIfAppropriate(); // Should not throw error
         }
+
+        [Fact]
+        public void RootSubCommandIsToolCommand()
+        {
+            var result = Parser.Instance.Parse("dotnetsay run -v arg");
+            result.RootSubCommandResult().Should().Be("dotnetsay");
+        }
     }
 }

--- a/src/Tests/dotnet.Tests/ParserTests/ToolRunParserTests.cs
+++ b/src/Tests/dotnet.Tests/ParserTests/ToolRunParserTests.cs
@@ -35,6 +35,7 @@ namespace Microsoft.DotNet.Tests.ParserTests
             var result = Parser.Instance.Parse("dotnet tool run dotnetsay hi");
             result.UnmatchedTokens.Should().Contain("hi");
             result.UnparsedTokens.Should().BeEmpty();
+            result.ShowHelpOrErrorIfAppropriate(); // Should not throw error
         }
 
         [Fact]
@@ -52,6 +53,7 @@ namespace Microsoft.DotNet.Tests.ParserTests
             var result = Parser.Instance.Parse("dotnet tool run dotnetsay hi1 -- hi2");
             result.UnmatchedTokens.Should().Contain("hi1");
             result.UnparsedTokens.Should().Contain("hi2");
+            result.ShowHelpOrErrorIfAppropriate(); // Should not throw error
         }
     }
 }

